modules/sor/sorV2/sorV2.service.ts

```diff
@@ -0,0 +1,243 @@
+    }
+
+    public mapResultToCowSwap(swap: string): string {
+        // TODO - match existing CowSwap SOR API format so its plug and play
+        return swap;
+    }
+
+    /**
+     * Gets a b-sdk Token based off tokenAddr.
+     * @param tokenAddr 
+     * @param chainId 
+     * @returns 
+     */
+    private async getToken(tokenAddr: Address, chainId: ChainId): Promise<Token> {
+        const tokens = await tokenService.getTokens();
+        const prismaToken = this.getPrismaToken(tokenAddr, tokens);
+        return new Token(
+            chainId,
+            tokenAddr,
+            prismaToken.decimals,
+            prismaToken.symbol,
+        );
+    }
+
+    private getPrismaToken(tokenAddress: string, tokens: PrismaToken[]): PrismaToken {
+        tokenAddress = tokenAddress.toLowerCase();
+        const match = tokens.find((token) => token.address === tokenAddress);
+
+        if (!match) {
+            throw new Error('Unknown token: ' + tokenAddress);
+        }
+        return match;
+    }
+
+    private mapSwapType(swapType: GqlSorSwapType): SwapKind {
+        return swapType === "EXACT_IN" ? SwapKind.GivenIn : SwapKind.GivenOut;
+    }
+
+    /**
+     * Fetch pools from Prisma and map to b-sdk BasePool.
+     * @returns 
+     */
+    private async getBasePoolsFromDb(): Promise<BasePool[]> {
+        const pools = await prisma.prismaPool.findMany({
+            where: { 
+                chain: networkContext.chain,
+                dynamicData: {
+                    totalSharesNum: {
+                        gt: 0.000000000001,
+                    },
+                    swapEnabled: true,
+                },
+            },
+            include: prismaPoolWithDynamic.include
+        });
+        const rawPools = this.mapToRawPools(pools);
+        // TODO - The pools below cause issue with b-sdk maths. Both have a token with balance = 0 so maybe that's issue?
+        const linearPoolsToIgnore = [
+            "0xbfa413a2ff0f20456d57b643746133f54bfe0cd20000000000000000000004c3", 
+            "0xdc063deafce952160ec112fa382ac206305657e60000000000000000000004c4", 
+        ]
+
+        const basePools = this.mapToBasePools(rawPools.filter(p => {
+            if (p.poolType == 'Linear') {
+                return !linearPoolsToIgnore.includes(p.id);
+            } else return true;
+        }));
+        // const basePools = this.mapToBasePools(rawPools);
+        return basePools;
+    }
+
+    /**
+     * Map Prisma pools to b-sdk RawPool.
+     * @param pools 
+     * @returns 
+     */
+    private mapToRawPools(pools: PrismaPoolWithDynamic[]): RawPool[] {
+        return pools.map(prismaPool => {
+            // b-sdk: src/data/types.ts
+            let rawPool: RawPool = {
+                id: prismaPool.id as Address,
+                address: prismaPool.address as Address,
+                poolType: this.mapRawPoolType(prismaPool.type),
+                poolTypeVersion: 1, // TODO - Can we add this to Prisma??
```

Comment:
Is it ok to add the pool type version to the database if its not already?

Reply:
We don't track the ID, but can be added of course. Question is where to source it from. I think balancer and beethoven use different versioning sometimes... Is that important for the SOR service?

Reply:
There's at least one case in new SOR where the maths differs depending on version.
I think having different versioning could make things more tricky. In the Balancer Subgraph there is the `poolTypeVersion` field. It is dependent on the pool factory.

Reply:
I will add versioning from the subgraph into the DB.